src/cryptoadvance/specter/util/descriptor.py

```diff
@@ -495,3 +495,26 @@ def sort_descriptor(descriptor, index=None):
     sorted_desc = desc.derive(index, keep_xpubs=True)
     sorted_desc.sort_keys = False
     return sorted_desc.serialize()
+
+
+def convert_receive_descriptor_to_combined_descriptor(descriptor: str) -> str:
+    # Split the descriptor by commas
+    parts = descriptor.split(",")
+
+    # Single sig descriptor
+    if len(parts) == 1:
+        single_sig_parts = descriptor.split("]")
+        single_sig_parts[1] = single_sig_parts[1].replace("/0/*", "/{0,1}/*")
+        combined_descriptor = "]".join(single_sig_parts)
+        return combined_descriptor
+
+    # Multisig descriptor
+    # Iterate through the parts and replace '/0/*' with '/{0,1}/*' if it occurs after an xpub
```

Comment:
Why so complicated? `*` only appears in derivations, so you can do the following:
`descriptor.split("#")[0].replace("/0/*", "/{0,1}/*")` - this will remove the checksum and replace all `/0/*` with `/{0,1}/*`.

Reply:
Addressed in https://github.com/cryptoadvance/specter-desktop/pull/2349/commits/e2c244415beecc1dc32b2b83729904278a1df6e7